Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -138,4 +171,68 @@ function isPrBuild(): boolean {
 
 function isNullOrEmpty(str) {
     return str === undefined || str === null || str.length === 0;
+}
+
+// Creates the string that comprises the build summary text, given the location of the /sonar build folder.
+function createSonarQubeBuildSummary(sqBuildFolder: string): string {
+    var taskReport: TaskReport = getSonarQubeTaskReport(sqBuildFolder);
+    if (!taskReport) {
+        throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
+    }
+
+    return util.format('[%s >](%s "%s Dashboard")',
+        tl.loc('sqAnalysisBuildSummaryLine_OneProject'), taskReport.dashboardUrl, taskReport.projectKey);
+}
+
+// Returns the location of the SonarQube integration staging directory.
+export /* public for test purposes */ function getSonarQubeStagingDirectory(): string {
```

Comment:
mkdirP as a convenience function is heavily used by other features that we own (PMD on Maven, for instance), and does not crash if the dir already exists.

Renaming the function - accepted.